src/adapter/src/catalog/open.rs

```diff
@@ -352,15 +367,21 @@ impl Catalog {
                 | BootstrapStateUpdateKind::SystemConfiguration(_)
                 | BootstrapStateUpdateKind::Cluster(_)
                 | BootstrapStateUpdateKind::NetworkPolicy(_)
-                | BootstrapStateUpdateKind::IntrospectionSourceIndex(_)
-                | BootstrapStateUpdateKind::ClusterReplica(_)
-                | BootstrapStateUpdateKind::SystemObjectMapping(_) => {
+                | BootstrapStateUpdateKind::ClusterReplica(_) => {
```

Comment:
Why was this re-shuffling needed? Is it because of how we measure/report timings below, for `pre_item_updates` and `sytem_item_updates`?

Reply:
We don't want to open the cache until we've loaded `SystemConfiguration`, because that contains the dyncfg that tells us whether or not the cache is enabled. However we want to use the cache to load `IntrospectionSourceIndex` and `SystemObjectMapping` because those contain system items that sometimes need to be optimized.

So splitting these up into two separate collections allows us to use the cache on system items.